src/database.ts

```diff
@@ -2101,7 +2103,16 @@ class Database extends common.GrpcServiceObject {
               session!.lastError = err;
               this.pool_.release(session!);
               this.getSnapshot(options, (err, snapshot) => {
-                span.end();
+                if (err) {
```

Comment:
@odeke-em Lets skip this change. In this we expect customer to invoke 'end' or 'error' event.  `isSessionNotFoundError` is not a very common error and this will also go away with our upcoming feature. So lets remove all this code. 
And move the `span.end()` outside of this.getSnapshot()  . Worst case in this error we will see 2 different spans.